### ISSUE TITLE
'css-loader' can't resolve the relative path in production mode since 14.6.0

### ISSUE THREAD

**Issue by SpicyBoyd** [vaadin/flow#11015]
### Description of the bug / feature
'css-loader' can't resolve the relative path in production mode.
### Minimal reproducible example

- My frontend folder looks like this:

```
-resources
  -frontend
    -css
        leaflet-map-styles.css
  -img
     map_layers_icon.png
```

- leaflet-map-styles.css

```css
[class="leaflet-control-layers-toggle"] {
	background-image: url('img/map_layers_icon.png') !important;
	background-size: 26px 26px;
}
```

- GoogleMap.java

```java
@CssImport(value = "./css/leaflet-map-styles.css", themeFor = "leaflet-map")
public class GoogleMap extends Div {...}
```

- addon: [Leaflet4Vaadin](https://vaadin.com/directory/component/leaflet4vaadin)

### Expected behavior
No problem when running/building image

### Actual behavior
```
ERROR in ../node_modules/@vaadin/flow-frontend/css/leaflet-map-styles.css
Module build failed (from ../node_modules/css-loader/dist/cjs.js):
Error: Can't resolve 'img/map_layers_icon.png' in 'D:\test\node_modules\@vaadin\flow-frontend\css'
    at D:\test\node_modules\enhanced-resolve\lib\Resolver.js:209:21
    at D:\test\node_modules\enhanced-resolve\lib\Resolver.js:285:5
    
    ...
    
@ ../target/frontend/generated-flow-imports-fallback.js 33:0-71 34:89-96
@ ../target/frontend/generated-flow-imports.js?babel-target=es6

...

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  18:38 min
[INFO] Finished at: 2021-05-18T13:23:50+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.vaadin:vaadin-maven-plugin:14.6.1:build-frontend (default) on project test: Webpack process exited with non-zero exit code.
[ERROR] Stderr: ''
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.vaadin:vaadin-maven-plugin:14.6.1:build-frontend (default) on project test: Webpack process exited with non-zero exit code.
Stderr: ''
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    
    ...    
```

### Versions:
    - Vaadin / Flow version: 14.6.1 (brokenness starts with 14.6.0)
    - Java version: 11
    - OS version: Windows 10

**Comment by caalador** [comment]
Could you give a minimal project that fails with this exception as with the given information I could not replicate the issue.

Is the theme file and image in the add-on location of `/src/main/resources/META-INF/resources/frontend/`?

**Comment by caalador** [comment]
It would seem that they are in the `META-INF/resources` folder as I got it to fail when moving them there.

As a workaround if you are not building an add-on you should be able to move the css and image to `{project_root}/frontend` and it should build fine.

**Comment by SpicyBoyd** [comment]
Yes, `/src/main/resources/META-INF/resources/frontend/`

**Comment by caalador** [comment]
For an add-on build you can as a workaround add into `webpack.generated.js` the line 
`flowDefaults.resolve.modules =  [...flowDefaults.resolve.modules, "node_modules\\@vaadin\\flow-frontend"];` 
before `module.exports = merge(flowDefaults, {`

**Comment by caalador** [comment]
Also having the url as relative would seem to works changing it to `background-image: url('../img/map_layers_icon.png') !important;`

**Comment by SpicyBoyd** [comment]
> For an add-on build you can as a workaround add into `webpack.generated.js` the line
> `flowDefaults.resolve.modules = [...flowDefaults.resolve.modules, "node_modules\\@vaadin\\flow-frontend"];`
> before `module.exports = merge(flowDefaults, {`

I am not a Vaadin Addon author.


> Also having the url as relative would seem to works changing it to `background-image: url('../img/map_layers_icon.png') !important;`

- `/src/main/resources/META-INF/resources/frontend/css/leaflet-map-styles.css`
- `/src/main/resources/META-INF/resources/img/map_layers_icon.png`

It's still not working.

**Comment by caalador** [comment]
If not making an add-on then I would suggest moving the resources according to [Table 1. Non-Spring projects](https://vaadin.com/docs/v14/flow/importing-dependencies/tutorial-ways-of-importing/#resource-cheat-sheet)

As for the relative path I didn't note that it was two folders up and not inside frontend.

**Comment by caalador** [comment]
My build actually doesn't fail if the image is in `/src/main/resources/META-INF/resources/img/map_layers_icon.png` and not under `/src/main/resources/META-INF/resources/frontend/img/map_layers_icon.png` would it be possible to have a simple sample project built on skeleton?

**Comment by SpicyBoyd** [comment]
> My build actually doesn't fail if the image is in `/src/main/resources/META-INF/resources/img/map_layers_icon.png` and not under `/src/main/resources/META-INF/resources/frontend/img/map_layers_icon.png` would it be possible to have a simple sample project built on skeleton that would work on 14.5, but fail on 14.6?
> 
> Is this a Spring project or a normal vaadin project?

https://github.com/SpicyBoyd/vaadin-flow-issue-11015

**Comment by SpicyBoyd** [comment]
After the project build

**Vaadin 14.5.5**

- `map_layers_icon.png` is at `target\classes\META-INF\resources\img\map_layers_icon.png`

**Vaadin 14.6.1**

- `map_layers_icon.png` is at `target\classes\META-INF\VAADIN\static\_\src\main\resources\META-INF\resources\img\map_layers_icon.png`

**Comment by Ortofta** [comment]
We are facing similar issues with our project when trying to update to 14.6.1. 

I was directed here from StackOverflow ([URL in CSS no longer works in Vaadin 14.6](https://stackoverflow.com/questions/67580595/url-in-css-no-longer-works-in-vaadin-14-6/67580678)), and asked to attach a demo project that illustrates the problem. 

The attached project illustrates the setup we have in our main project, where things are separated into different modules. It's not complete in any way, and it probably isn't even deploy-able - but it demonstrates the issue we are having at build time.

Running `mvn vaadin:prepare-frontend` and `mvn vaadin:build-frontend` in the WAR-module succeeds with Vaadin 14.5.5, but fails in the build-frontend step with Vaadin 14.6.1:

`ERROR in ../node_modules/@vaadin/flow-frontend/styles/components/password-field.css
Module build failed (from ../node_modules/css-loader/dist/cjs.js):
Error: Can't resolve '../images/ic-password-show.svg' in 'C:\data\junk\example\demo\war-module\node_modules\@vaadin\flow-frontend\styles\components'
    at C:\data\junk\example\demo\war-module\node_modules\enhanced-resolve\lib\Resolver.js:209:21
    at C:\data\junk\example\demo\war-module\node_modules\enhanced-resolve\lib\Resolver.js:285:5
    at eval (eval at create (C:\data\junk\example\demo\war-module\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:15:1)
    at C:\data\junk\example\demo\war-module\node_modules\enhanced-resolve\lib\UnsafeCachePlugin.js:44:7
    at C:\data\junk\example\demo\war-module\node_modules\enhanced-resolve\lib\Resolver.js:285:5
    at eval (eval at create (C:\data\junk\example\demo\war-module\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:15:1)
    at C:\data\junk\example\demo\war-module\node_modules\enhanced-resolve\lib\Resolver.js:285:5
    at eval (eval at create (C:\data\junk\example\demo\war-module\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:27:1)
    at C:\data\junk\example\demo\war-module\node_modules\enhanced-resolve\lib\DescriptionFilePlugin.js:67:43
    at C:\data\junk\example\demo\war-module\node_modules\enhanced-resolve\lib\Resolver.js:285:5
    at eval (eval at create (C:\data\junk\example\demo\war-module\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:16:1)
    at C:\data\junk\example\demo\war-module\node_modules\enhanced-resolve\lib\Resolver.js:285:5
    at eval (eval at create (C:\data\junk\example\demo\war-module\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:27:1)
    at C:\data\junk\example\demo\war-module\node_modules\enhanced-resolve\lib\DescriptionFilePlugin.js:67:43
    at C:\data\junk\example\demo\war-module\node_modules\enhanced-resolve\lib\Resolver.js:285:5
    at eval (eval at create (C:\data\junk\example\demo\war-module\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:16:1)
    at C:\data\junk\example\demo\war-module\node_modules\enhanced-resolve\lib\Resolver.js:285:5
    at eval (eval at create (C:\data\junk\example\demo\war-module\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:15:1)
    at C:\data\junk\example\demo\war-module\node_modules\enhanced-resolve\lib\DirectoryExistsPlugin.js:27:15
    at C:\data\junk\example\demo\war-module\node_modules\enhanced-resolve\lib\CachedInputFileSystem.js:85:15
    at processTicksAndRejections (internal/process/task_queues.js:75:11)
 @ ../target/frontend/generated-flow-imports-fallback.js 7:0-80 8:109-115
 @ ../target/frontend/generated-flow-imports.js?babel-target=es5`

[demo.zip](https://github.com/vaadin/flow/files/6513611/demo.zip)

**Comment by gero1992** [comment]
Having the same problem, when trying to migrate to migrate to vaadin 14.6.1

Having a spring boot project (maven parent project used as a base for other applications) with the following structure:

- resources
  - META-INF
    - resources
       - fonts
          - some.woff
       - frontend
          - styles
            - fonts.css


fonts.css contains:

```
@font-face {
    font-family: 'some';
    src: url('fonts/some.woff') format('woff');
    ...
}
```

The font.css is imported in the MainLayout with:

```
@CssImport(value = "./styles/fonts.css")
```

When I start the application I get:

`
LOC - 25.05.2021 16:49:50.948 [webpack] ERROR dev-webpack - ERROR in ../node_modules/@vaadin/flow-frontend/styles/fonts.css
LOC - 25.05.2021 16:49:50.948 [webpack] ERROR dev-webpack - Module build failed (from ../node_modules/css-loader/dist/cjs.js):
LOC - 25.05.2021 16:49:50.948 [webpack] ERROR dev-webpack - Error: Can't resolve '../fonts/some.woff' in 'C:\project-ui\node_modules\@vaadin\flow-frontend\styles'
`

Before moving to vaadin 14.6.x it was working fine. I also don't have the need to create a custom theme, so I would be happy if there is a solution without requiring that

**Comment by caalador** [comment]
The fix will come in 14.6.2

Closing issue as the fix has been merged and it should roll out sometime soon.

Until release of 14.6.2 you can in the mean while add the raw-loader dependency to a java class with
`@NpmPackage(value = "raw-loader", version = "3.1.0")`
and then adding to `webpack.config.js` the lines
```
if(flowDefaults.module.rules[2].test.toString().includes('.css')) {
  flowDefaults.module.rules[2].use = [ {loader: 'raw-loader' }];
} else if(flowDefaults.module.rules[1].test.toString().includes('.css')) {
  flowDefaults.module.rules[1].use = [ {loader: 'raw-loader' }];
}
```
